### PR TITLE
Flight Gear visualisation with native fdm

### PR DESCRIPTION
**Pull request by podhrmic**
Allows using NPS with Flight Gear's Native FDM. As a result, it is possible to show moving control surfaces (flaps, ailerons etc) and spinning propeller. 

Simulator has to be started with ``--fg_fdm 1`` option, and flight gear with ``--native-fdm`` option (instead of ``--native_gui``). 

For example NPS launch command:
```./sw/simulator/pprzsim-launch -a Bixler -t nps --js_dev 0 --fg_host 127.0.01 --fg_fdm 1```

And Flight Gear launch command:
```fgfs --fdm=null --native-gui=socket,in,30,,5501,udp --prop:/sim/model/path=Models/Aircraft/paparazzi/easystar.xml```

More about Native FDM for Flight Gear here: http://wiki.flightgear.org/index.php/Property_Tree/Sockets

### CODE REVIEW

**Comment by flixr** [comment]
nice! Can you plz also add the `--fg_fdm` option to the help/usage

File: sw/simulator/nps/nps_main.c [hunk 1]
```diff
@@ -169,8 +170,14 @@ static void nps_main_run_sim_step(void) {
 static void nps_main_display(void) {
   //  printf("display at %f\n", nps_main.display_time);
   nps_ivy_display();
-  if (nps_main.fg_host)
-    nps_flightgear_send();
+  if (nps_main.fg_host) {
+    if (nps_main.fg_fdm) {
+          nps_flightgear_send_fdm();
```
Comment (flixr) [inline sw/simulator/nps/nps_main.c]: intentation....

File: sw/simulator/nps/nps_main.c [hunk 2]
```diff
@@ -311,6 +319,7 @@ static bool_t nps_main_parse_options(int argc, char** argv) {
       {"rc_script", 1, NULL, 0},
       {"ivy_bus", 1, NULL, 0},
       {"time_factor", 1, NULL, 0},
+      {"fg_fdm", 1, NULL, 0},
```
Comment (flixr) [inline sw/simulator/nps/nps_main.c]: would make more sense to not require an arg for `fg_fdm`, so `{"fg_fdm", 0, NULL, 0}`

File: sw/simulator/pprzsim-launch [hunk 3]
```diff
@@ -78,6 +78,8 @@ def main():
                         help="Number of RC script to use")
     nps_opts.add_option("--time_factor", type="float", action="store", metavar="factor",
                         help="Time factor (default 1.0)")
+    nps_opts.add_option("--fg_fdm", type="int", action="store", metavar="factor",
```
Comment (flixr) [inline sw/simulator/pprzsim-launch]: no reason to require an arg here... so `"--fg_fdm", action="store_true",` would be nicer

File: sw/simulator/pprzsim-launch [hunk 4]
```diff
@@ -143,6 +145,9 @@ def main():
         if options.time_factor:
             simargs.append("--time_factor")
             simargs.append(str(options.time_factor))
+        if options.fg_fdm:
+            simargs.append("--fg_fdm")
+            simargs.append(str(options.fg_fdm))
```
Comment (flixr) [inline sw/simulator/pprzsim-launch]: and then remove that line

File: sw/simulator/pprzsim-launch [hunk 5]
```diff
@@ -78,6 +78,8 @@ def main():
                         help="Number of RC script to use")
     nps_opts.add_option("--time_factor", type="float", action="store", metavar="factor",
                         help="Time factor (default 1.0)")
+    nps_opts.add_option("--fg_fdm", type="int", action="store", metavar="factor",
+                        help="FlightGear FDM")
```
Comment (flixr) [inline sw/simulator/pprzsim-launch]: help could be a bit more descriptive ;-)
e.g. "Use FlightGear native-fdm protocol instead of native-gui"

**Comment by flixr** [comment]
Just curious, do you know why native-fdm needs to have the struct in network order and native-gui not??

File: sw/simulator/nps/nps_main.c (hunk 1 above)
Comment (podhrmic) [inline sw/simulator/nps/nps_main.c]: fixed with ``fix_code_style.sh``

File: sw/simulator/nps/nps_main.c (hunk 2 above)
Comment (podhrmic) [inline sw/simulator/nps/nps_main.c]: fixed

File: sw/simulator/pprzsim-launch (hunk 3 above)
Comment (podhrmic) [inline sw/simulator/pprzsim-launch]: fixed

File: sw/simulator/pprzsim-launch (hunk 4 above)
Comment (podhrmic) [inline sw/simulator/pprzsim-launch]: fixed

File: sw/simulator/pprzsim-launch (hunk 5 above)
Comment (podhrmic) [inline sw/simulator/pprzsim-launch]: fixed

**Comment by podhrmic** [comment]
I think it was either sheer luck (native-gui just happened to work with the same byte order), or FlightGear internally treats the native-gui differently than native-fdm...

File: sw/simulator/pprzsim-launch [hunk 6]
```diff
@@ -78,6 +78,8 @@ def main():
                         help="Number of RC script to use")
     nps_opts.add_option("--time_factor", type="float", action="store", metavar="factor",
                         help="Time factor (default 1.0)")
+    nps_opts.add_option("--fg_fdm", action="store_true", metavar="factor",
```
Comment (flixr) [inline sw/simulator/pprzsim-launch]: metavar doesn't make any sense for an option without arguments anymore...

**Comment by flixr** [comment]
Also please if you reformat a whole file (that has not conformed before), it is much nicer to separate the functional changes and the pure reformatting in separate commits...
Then you can later easily see what the actual changes where when going through the log...

**Pull request closed without merging**